ql/src/java/org/apache/hadoop/hive/ql/optimizer/stats/annotation/StatsRulesProcFactory.java

```diff
@@ -1734,7 +1735,7 @@ private boolean checkMapSideAggregation(GroupByOperator gop,
         float hashAggMaxThreshold = conf.getFloatVar(HiveConf.ConfVars.HIVEMAPAGGRMEMORYTHRESHOLD);
 
         // get available map memory
-        long totalMemory = StatsUtils.getAvailableMemory(conf, true) * 1000L * 1000L;
+        long totalMemory = DagUtils.getContainerResource(conf).getMemorySize() * 1000L * 1000L;
```

Comment:
Existing issue, but please address here: this should be in MiB (1024L)

Reply:
ACK